packages/mui-utils/src/useTimeout/useTimeout.ts

```diff
@@ -1,6 +1,6 @@
 'use client';
-import useLazyRef from '../useLazyRef/useLazyRef';
-import useOnMount from '../useOnMount/useOnMount';
+import useLazyRef from '../useLazyRef';
+import useEnhancedEffect from '../useEnhancedEffect';
```

Comment:
It's not the first time I see `useEnhancedEffect` cause such issue, the hook makes it less apparent that `useLayoutEffect` is used, and that it runs on a different cycle than `useEffect`.